### PR TITLE
E6+V: Vampirism, addons, integrations, and recipes for E6

### PR DESCRIPTION
**Pull request by yagoplx**
This adds the following mods to Enigmatica 6:
[Vampirism](https://www.curseforge.com/minecraft/mc-mods/vampirism-become-a-vampire)
[Vampirism Integrations](https://www.curseforge.com/minecraft/mc-mods/vampirism-integrations)
[Vampires Need Umbrellas](https://www.curseforge.com/minecraft/mc-mods/vampires-need-umbrellas)
[Guide-API Village and Pillage](https://www.curseforge.com/minecraft/mc-mods/guide-api-village-and-pillage)

**Questions you may have**
Q: Why Vampirism?
A: It's a mod I like and I'd like to have more people get to know it - Enigmatica is influential in the modded minecraft space so it's my best bet on making that happen. Notably it has the trait of being a very self-centered mod - but it doesn't need to be open. It's got enough content by itself and the player should have the final choice on whether they want to explore the mod's factions and suffer/enjoy the gameplay twists it brings.

Working branch: https://github.com/yagoplx/Enigmatica6/tree/develop

**Changes so far**
* Vampires are now slightly rarer than witches. You can still find a bunch of them on the vampire biome. (if you can get there)
* Changes to vampire balance
   -  level 1 vampires cannot walk in the sun without any ill effect, now get weakness
   -  slightly decreased vampire blood consumption, due to "true passive" animals being rarer in E6. Should make starving less common early on.
* Vampire players now only need to wait 30 seconds instead of 60 to return from death in a severely weakened form. Cooldown unchanged.
* Added Vampirism guidebook to akashic tome reward & recipe. Akashic tomes can now contain the guidebook. The guidebook will no longer be given at new world or new player.
* Made hunter camps somewhat rarer
* Added a recipe for empty blood bottles
* Added 2 tips. One is helpful, the other is a hearty reminder of the existence of garlic bread. This food item is obviously more important than everything else in the entire pack.
* "Un-creativefied" sunscreen beacons (normal mode + expert)
* Vampire fangs are craftable (normal mode + expert)
* Changed some progression recipes for hunters and vampires on expert
* Removed vampire accessories from normal loot chests. It's exclusive to vampires, and feels out of place in E6 (uses its own GUI). Vanity stuff to wear on armor slots whilst benefitting from natural vampire armor is kept.
* Removed some fluff (lord title in chat)
* Adds a single icon to the Adventure tab
* You can now cast TiC blood into vampirism bottles

To do checklist: 
* :heavy_check_mark:  Add mod configuration files with tweaks (a430336416254be665313b04b688f15a595883b7)
* :heavy_check_mark:  Add structure generation rules to Terraforged (a430336416254be665313b04b688f15a595883b7)
* :heavy_check_mark:  Fix faction button overlaying diet button (a430336416254be665313b04b688f15a595883b7)
* :heavy_check_mark:  Fix transformation being cancelled by bed (d1ab9a5ce05f352c2de3e1f1f10658eada54f5f6)
* :heavy_check_mark:  Unificate Tinker's blood with vampirism blood, or add a way to convert between (https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/f020f62a1e5f3f143755dfdb633528e4f6c69c60)
* :heavy_check_mark:  Add biome generation rules to Terraforged (https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/bee84c3ee9f2c0b7dfab8113a9b2fd955a1acfdc)
* :heavy_check_mark:  E6 KubeJS integration (a3a4f2e4144516709384099ad2b6afd31bad28fa, da2805187ed59fb4af5fa8ebd168c0265c9da9d2, https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/6b4c2d161e09a0f8bb8fdb40b4ea3c83b874780f, https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/94520cf71724db12a19db44f2f776af95a0f687c)
* :heavy_check_mark:  Add Vampirism-themed tips (https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/a6a6f7011b58d35a4c32e53f2d871224870b27f8)
* 🕗  Make vampire players uneligible for diet buffs (https://github.com/TeamLapen/VampirismIntegrations/issues/48)
  - Negligible impact on gameplay, this feature is a polishment and can wait in the back burner.
* :heavy_check_mark:  Address Niels' concern: make vampires rarer - but it's so that you don't just treat them as just yet another backyard pest - instead I want it to be seen as a rare opportunity to be turned. However once the Nether is available you will be able to craft the item to turn yourself. (5a9943435a0c1a4321f73ee3baf29b451c05c951, https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/857788c45bae5827a64fb66b137aeee782b70aaa)
* :heavy_check_mark: Draft up a quest line for Vamp and implement it (https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/639297ea6936b73e7d0f1eac20da8b098a3a5467)
* :heavy_check_mark: Expert mode recipes for Vamp & addon. (https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/ccdbc45e76d9475d3d5b30521e495eb59ba92c5d, https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/94520cf71724db12a19db44f2f776af95a0f687c)
* :heavy_check_mark:  Add garlic bread and garlic crop to integration (https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/87065dab3cb955941570c6a0db002fcf1b6e8acc)
* :heavy_check_mark: Remove obi belt, ring, amulet and fangs from vanilla loot tables (https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/32f585f5bd5a7067d9b651ecdd4bf1545eba8a41)

* :heavy_check_mark:  Ask for feedback

To do post-completion:
* :heavy_check_mark:  Package E6+V as a separate "pack addon" so that people can play around with it right now
* ❌  Actually add the mods & changes to the pack metadata, source & official packages

Current Status: 
* 📦 Complete
   - Package: 0.5.2

### CODE REVIEW

**Comment by yagoplx** [comment]
This is my first time with KubeJS, so there may be some oopsies I didn't catch but so far so good, no kubejs errors or warnings.

My thing with vampirism on the pack is exactly having it be a side mod for people who want a gameplay twist. Yes- vampirism can be a side mod. Just look at MC Eternal :)

This took quite some energy and thus I suspect that the thing above is faulty- well it doesn't seem to me, because I'm tired, but hopefully tomorrow I will be able to find many flaws on it and fix them.

As is customary, yes, I'm eager to hear suggestions and constructive feedback, and I'd like to know about things I have missed on integration that you feel are important.

File: kubejs/server_scripts/enigmatica/kubejs/base/recipes/shapeless.js [hunk 1]
```diff
@@ -553,7 +564,8 @@ onEvent('recipes', (event) => {
         },
         {
             output: Item.of('minecraft:string', 3),
-            inputs: ['#forge:crops/flax', '#forge:crops/flax', '#forge:crops/flax']
+            inputs: ['#forge:crops/flax', '#forge:crops/flax', '#forge:crops/flax'],
+            id: 'simplefarming:string'
```
Comment (yagoplx) [inline kubejs/server_scripts/enigmatica/kubejs/base/recipes/shapeless.js:568]: I'm not sure what happened here but it seems... benign?

**Comment by yagoplx** [comment]
> x Remove obi belt, ring, amulet from vanilla loot tables (help needed)

Yikes, I don't want these occupying space in what is essentially a ton of loot chests.
Basically vampirism injects its items into the vanilla loot tables [via this code](https://github.com/TeamLapen/Vampirism/blob/320e6240f0c2921d94e137d29305c1f36602610f/src/main/java/de/teamlapen/vampirism/core/ModLootTables.java#L34-L38). However I need to remove the amulet, obi belt and ring accessories from these tables if I want this to work out.

Does anybody know how I'd go about this? I wonder if we could use KubeJS as usual (I have no clue how other than fully replacing the loot table json which is uncool) but my primary clue is making an issue on Vampirism to include an option to disable modification of vanilla loot tables.

**Comment by yagoplx** [comment]
Here is a neat package containing all that was changed. You can extract this on top of your E6 instance to test out the changes and how you feel about the impact. Feedback is always appreciated. For now the initial implementation of "somewhat integrating Vampirism into the E6 scene" is complete.

The aforementioned four mods are included as well.
Should work on ` develop ` and `0.5.2`.

[E6plusV-0.1.zip](https://github.com/NillerMedDild/Enigmatica6/files/6743941/E6plusV-0.1.zip)
SHA1: 137439819b6e00e76686154078e1b04788750b3b
Contains up to commit: https://github.com/NillerMedDild/Enigmatica6/pull/2582/commits/51a8856677b54946b09a370fd84f44d338df02ad

**Comment by Hesitate-d** [comment]
Will this mod actually be added to E6 if so i'd really look forward to it as it is one of my favourite and most enjoyable mods and i'd like a more updated version of it as E6 has with the rest of the mods as i've been stuck on an outdated version on MC eternal  and i would really enjoy this mod to be put into this modpack so i can swapover from Eternal to E6.

**Comment by MuteTiefling** [comment]
The mod was already declined here: https://github.com/NillerMedDild/Enigmatica6/issues/2454

It's not being added. And for the record, it was declined before all of this work went into it. 

At the end of the day, we can't add every mod, as much as some people might enjoy seeing them. You are, of course, free to add whatever mods you like to a pack, but understand that Enigmatica 6 is still under constant development and our changes may well break whatever you've added. Such breakages will not be supported. 

Yago, I'd give my feedback on your work here, but.. honestly I don't want to spend the time reviewing your changes when you've already been told no twice. Thank you for your suggestion just the same, but at this time Vampirism has no future in E6.

**Comment by yagoplx** [comment]
Heh, yes, a lot of MC Eternal players are gonna resonate with this one. In that pack, Vampirism is completely regarded as a side mod just to give the gameplay that little twist and **it just works so well.**

**Comment by Hesitate-d** [comment]
It's a shame but i guess that's how it be.

**Comment by yagoplx** [comment]
It's not the end of the line @Hesitate-d. I will continue maitaining the addon as I have the skills and time for it.

[You are welcome to send suggestions and bug reports related to the integration here.](https://github.com/yagoplx/E6-V-Issue-Tracker)

**Comment by Hesitate-d** [comment]
Is there any latest modpack that have vampirism up to date? Cause i really wanna play it and i've hit an end with MC eternal as its kinda outdated?

**Pull request closed without merging**